homeassistant/components/recorder/migration.py

```diff
@@ -17,6 +17,17 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def raise_if_exception_missing_str(ex, match_substrs):
+    """Raise an exception if the exception and cause do not contain the match substrs."""
+    lower_ex_strs = [str(ex).lower(), str(ex.__cause__).lower()]
+    for str_sub in match_substrs:
+        for exc_str in lower_ex_strs:
+            if exc_str and str_sub in exc_str:
```

Comment:
Is this if-check to make sure `exc_str` is not empty? But that doesn't matter much unless `str_sub` is empty too?

Reply:
Yes. It is. I'll adjust it when I get back home

Reply:
Looked at the code, it's just a couple of iterations, it's all good